### ISSUE TITLE
Build error on Appveyor with recent changes.

### ISSUE THREAD

**Issue by Ky7m** [dotnet/BenchmarkDotNet#530]
After recent changes there is build issue on Windows platfrom:
```
========================================
FastTests
========================================
Executing task: FastTests
Test run for C:\projects\benchmarkdotnet\tests\BenchmarkDotNet.Tests\bin\Release\netcoreapp1.1\BenchmarkDotNet.Tests.dll(.NETCoreApp,Version=v1.1)
Microsoft (R) Test Execution Command Line Tool Version 15.0.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.
Starting test execution, please wait...
Illegal characters in path.
Parameter name: path
```
It is also reproducible on local environment with the latest changes:
```
dotnet.exe test ".\tests\BenchmarkDotNet.Tests\BenchmarkDotNet.Tests.csproj" --configuration Release --no-build
```
If I remove `--no-build` argument error doesn't appear. **--no-build** flag means that do not build the test project prior to running it. It was added as optimization to save time because we build entire solution on previous step.
Without `--no-build` everything works as expected:
```
dotnet.exe test ".\tests\BenchmarkDotNet.Tests\BenchmarkDotNet.Tests.csproj" --configuration Release
```

@adamsitnik what do you think is it a root of build issue, are there some structures/configuration changes that lead to this strange behavior?

Are you ok to remove `--no-build` argument? If yes, I will send a new PR to fix build on AppVeyor // cc: @AndreyAkinshin

**Comment by adamsitnik** [comment]
@Ky7m big thanks for spotting this!

I am quite surprised with this error: my changes went [green](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/build/0.10.9.281) and after this we started getting [red](https://ci.appveyor.com/project/dotnetfoundation/benchmarkdotnet/build/0.10.9.282-imppaqbr) ones. I am suprised because we get a timeout on AppVeyour:

> Build execution time has reached the maximum allowed time for your plan (60 minutes).

I would expect the build to fail fast with this error (Illegal characters in path).

What I changed:
I added two new projects and set their build output to another project's resources (to embed one exe in another). It should not be a problem.

I would be very happy to accept the PR that removes `--no-build` if it helps. It seems to me like an AppVeyor or dotnet cli bug.
